controllers/provision_pod_test.go

```diff
@@ -0,0 +1,250 @@
+package controllers
+
+//	no.	config	accessConfig	secret	pod	updated	state
+//	1	x	x		x	x
+//	2	o	x		x	x
+//	3	o	o		x	x
+//	4	o	o		o	x	no
+//	5	o	o		o	x	yes
+//	6	o	o		o	o	no	Completed
+//	7	o	o		o	o	no	Running
+
+import (
+	"context"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	hypersdsv1alpha1 "github.com/tmax-cloud/hypersds-operator/api/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+var _ = Describe("syncProvisioner", func() {
+	Context("1. with no resource", func() {
+		r := createFakeCephClusterReconciler()
+		err := r.syncProvisioner()
+
+		It("Should not return error", func() {
+			Expect(err).Should(BeNil())
+		})
+	})
+
+	Context("2. with configmap", func() {
+		cm := &corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      getConfigMapName(testCephClusterName),
+				Namespace: testCephClusterNs,
+			},
+		}
+		r := createFakeCephClusterReconciler(cm)
+		err := r.syncProvisioner()
+
+		It("Should not return error", func() {
+			Expect(err).Should(BeNil())
+		})
+	})
+
+	Context("3. with configmap and access configmap", func() {
+		cm := &corev1.ConfigMap{
```

Comment:
Maybe to have functions for creating each fake object like cm, accessCM (updated no/yes), and secret and put it all in `test-util.go` is helpful?